src/isNotNil.js

```diff
@@ -1,6 +1,5 @@
-const { complement, isNil } = require('ramda')
-
 // isNotNil :: a -> Boolean
-const isNotNil = complement(isNil)
+const isNotNil =
+  x => x != null
```

Comment:
`x => x != null && !Number.isNaN(x)` ?

Reply:
There isn't support for `Number.isNaN` in IE, so either a polyfill or a don't-use-it-if-it's-not-around may be in order.

Reply:
per https://devdocs.io/javascript/global_objects/number/isnan

Reply:
and the polyfill suggested at that link should make you smile.  it made me smile, at least.

Reply:
https://github.com/ramda/ramda/blob/master/source/isNil.js

Ramda does not account for NaN. So IMHO, we shouldn't either.

Reply:
We could add a `isNaN` to funky as part of the api or as an internal helpers:

```js
const isNaN =
    x => x !== x
```

Reply:
@mgreystone I am of the opinion that we should. While `typeof NaN` now I have a GOOD reason why we should by it is so math-y, please let me know if you want me to go into depth about any part of this.

While `typeof NaN === "number"`, it actually behaves as what is called a Terminal Object in our Category of JS. If you look at the connections it can make with other Objects (types) in our Category. As it is a Terminal Object (Basically a singleton set for our category), there is a unique isomorphism between all Terminal Types in a given category. Because we can jump between them with that isomorphism it really is just a matter of changing labels.

Reply:
If we do end up accounting for `NaN` in this function. To avoid addfitional stack frames and trips into `node_modules` when debugging a stack trace, this *could* be:

```js
const isNotNull = x =>
  !(x !== x || x == null)
```

Reply:
Yeah, i get that (not really). But since we primarily use this as a ramda companion, i find this confusing:

```js
R.isNil(NaN) // -> false
funky.isNotNil(NaN) // -false
```

😕

Reply:
I'd be more likely to flip it, since `null` and `undefined` are slightly more frequent than `NaN`, and JS will just skip the right side in those cases:
```js
const isNotNull = x =>
  !(x == null || x !== x)
```

Reply:
Was just about to mention that @flintinatux.

Reply:
@mgreystone That is a valid point. Could get 🌽fusing if that is all funky is, just a paired down `ramda-extended`

Reply:
Should we have a vote on this since it seems polarizing?

Reply:
Alright I am going to change it back to `isNotNil(NaN) === true`

Reply:
To answer the results you are seeing `{ a: true } !== { a: true }` BUT

```js
const x = { a: true }

x !== x
//=> false
```

Reply:
Thank goodness you brought up the people aspect @mgreystone, because I made (2) claims:
* All Terminal Objects have a unique isomorphisim between them
* `NaN` is a Terminal Object in the Category of JavaScript

and BOTH proofs are super duper intense.